src/main/java/org/folio/rest/impl/util/OkapiResponseUtil.java

```diff
@@ -0,0 +1,47 @@
+package org.folio.rest.impl.util;
+
+import io.vertx.core.AsyncResult;
+
+import javax.ws.rs.core.Response;
+
+import org.folio.rest.jaxrs.model.Error;
+import org.folio.rest.jaxrs.model.Errors;
+
+/**
+ * Utility methods for Okapi Responses.
+ */
+public class OkapiResponseUtil {
+
+  private OkapiResponseUtil() {
+    throw new UnsupportedOperationException("Do not instantiate");
+  }
+
+  public static String getErrorMessage(AsyncResult<Response> reply) {
+    String message = null;
+
+    if (reply.succeeded() && reply.result().getStatus() >= 400
+      && reply.result().getStatus() < 500 && reply.result().hasEntity()) {
+
+      // When entity is an instance of Errors, the getEntity().toString()
+      // will return an object identifier. Process the object to correctly
+      // parse the message.
+      if (reply.result().getEntity() instanceof Errors) {
+        Errors errors = (Errors) reply.result().getEntity();
+
+        for (int i = 0; i < errors.getErrors().size(); i++) {
```

Comment:
```
if (errors.getErrors().size() > 0) {
  Error error = errors.getErrors().get(0);
  message = error.getMessage().toLowerCase();
}
```

Reply:
We only ever expect one Error in the Errors list?

Reply:
I think yes. Alternatively we can concatenate all of them, or return an array. 

I just spotted that you're rewriting the `message` on each iteration, we can simplify it somehow.

Reply:
> I just spotted that you're rewriting the message on each iteration, we can simplify it somehow.

Oops, I must have forgotten that logic when I moved the `contains()` check outside of the loop.
As it stands, my logic there is unsound and needs to be fixed.
The concatination may be a good way to catch all of the possibilities.

Reply:
There is no guarantee that there will only be one error. We should likely forward on any errors we receive from PgUtil